emcc.py

```diff
@@ -93,6 +95,10 @@
     # wasm-ld doesn't support soname or other dynamic linking flags (yet).   Ignore them
     # in order to aid build systems that want to pass these flags.
     '-soname': True,
+    # there is no need to specify --gc-sections: we add it automatically when
+    # it is possible, and not when it would be an error
+    '--gc-sections': False,
```

Comment:
Can you land this without `--gc-sections` and I'll take a look at the `--relocatable` issue.